### ISSUE TITLE
Cryptography has done a release which breaks our installs

### ISSUE THREAD

**Issue by mikalstill** [shakenfist/shakenfist#686]
```
Collecting cryptography (from ansible-base<2.11,>=2.10.5->ansible)
  Downloading https://files.pythonhosted.org/packages/06/ed/cb79cc94ec58d9d92557238fc6c629cd6e07d72334d2de556aecc2211370/cryptography-3.4.1.tar.gz (544kB)
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-t2d8qcjs/cryptography/setup.py", line 14, in <module>
        from setuptools_rust import RustExtension
    ModuleNotFoundError: No module named 'setuptools_rust'
    
            =============================DEBUG ASSISTANCE==========================
            If you are seeing an error here please try the following to
            successfully install cryptography:
    
            Upgrade to the latest pip and try again. This will fix errors for most
            users. See: https://pip.pypa.io/en/stable/installing/#upgrading-pip
            =============================DEBUG ASSISTANCE==========================
    
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-t2d8qcjs/cryptography/
```

**Comment by mikalstill** [comment]
We need to decide if we pin to something older, or work out how to add the rust thing.

**Comment by mandoonandy** [comment]
I'd be comfortable pining it, but I would hope that the issue is fixed within 24 hours?

**Comment by jackadamson** [comment]
After looking at the CI instance, it looks like it could be due to pip3 being a binary provided by python3-pip and not using the lastest version of pip.

```
jack@sfci-2uzs1f:~$ pip3 -V
pip 9.0.1 from /usr/lib/python3/dist-packages (python 3.6)
```

Possibly worth changing to Ubuntu 20.04 if we aren't intending to support 18.04, or using `pip` instead of `pip3`

**Comment by mikalstill** [comment]
Jack did some spelunking, which caused me to notice that we're running 18.04 in our CI not the 20.04 I thought we were running. We think this might be because that is causing us to run and older pip than we expected. I'll take a swing at that ASAP.

**Comment by lod** [comment]
Difficult tradeoff
Pinning leaves you lagging a security package, though mostly a wrapper for the underlying libssl
Tracking will hit your platform capability

There is a requirement for rust >= 1.45
https://github.com/pyca/cryptography/issues/5769

Debian stable: 1.41
Debian testing: 1.48
Ubuntu 20.04 (amd64/i386): 1.47
Ubuntu 20.04 (all others): 1.43
Ubuntu 18.04 (amd64/i386): 1.47
Ubuntu 18.04 (all others): 1.24
Fedora updates: 1.47 >= f31
Fedora base f33 (latest): 1.46
Centos 8: 1.45

**Comment by mikalstill** [comment]
This is now resolved by moving CI to Ubuntu 20.04.
